app/models/library_entry.rb

```diff
@@ -134,5 +142,20 @@ def activity
       user.update_feed_completed!
       user.update_profile_completed!
     end
+
+    # Sync MAL updates if linked profile exists
+    if sync_to_mal?
+      return unless media_type.in? %w[Anime Manga]
```

Comment:
Now that this is no longer named `linked_profile?`, maybe move this check into `sync_to_mal?`

Then it's just

```ruby
MyAnimeListSyncWorker.perform_async(self, 'create/update') if sync_to_mal?
```